src/pytorch_lightning/loops/fit_loop.py

```diff
@@ -170,20 +176,10 @@ def done(self) -> bool:
             rank_zero_info(f"`Trainer.fit` stopped: `max_epochs={self.max_epochs!r}` reached.")
             return True
 
-        if self.trainer.should_stop:
-            # early stopping
-            met_min_epochs = self.epoch_progress.current.processed >= self.min_epochs if self.min_epochs else True
-            met_min_steps = self.epoch_loop.global_step >= self.min_steps if self.min_steps else True
-            if met_min_epochs and met_min_steps:
-                self.trainer.should_stop = True
-                rank_zero_debug("`Trainer.fit` stopped: `trainer.should_stop` was set.")
-                return True
-            else:
-                rank_zero_info(
-                    f"Trainer was signaled to stop but the required `min_epochs={self.min_epochs!r}` or"
-                    f" `min_steps={self.min_steps!r}` has not been met. Training will continue..."
-                )
-        self.trainer.should_stop = False
+        if self.trainer.should_stop and self._should_stop_early:
+            rank_zero_debug("`Trainer.fit` stopped: `trainer.should_stop` was set.")
```

Comment:
> Should show the same message here?

I don't think so because they have different meanings. That tells the training will continue and this one tells that training will terminate.

> Is there an edge case where both this one and the one in training epoch loop would appear?

no because when the message in training epoch loop occurs like this:
```
if not limit_reached:
    show info message and continue training epoch loop.
```
so it will not come back to fit loop.